echion/frame.h

```diff
@@ -450,50 +455,77 @@ Frame &Frame::read(PyObject *frame_addr, PyObject **prev_addr)
 
 #if PY_VERSION_HEX >= 0x030b0000
 // ------------------------------------------------------------------------
-Frame &Frame::read_local(_PyInterpreterFrame *frame_addr, PyObject **prev_addr)
+Frame &Frame::read_with_stack_chunk(_PyInterpreterFrame *frame_addr, PyObject **prev_addr)
 {
+    _PyInterpreterFrame iframe;
 #if PY_VERSION_HEX >= 0x030d0000
     // From Python versions 3.13, f_executable can have objects other than
     // code objects for an internal frame. We need to skip some frames if
     // its f_executable is not code as suggested here:
     // https://github.com/python/cpython/issues/100987#issuecomment-1485556487
     PyObject f_executable;
 
-    for (;frame_addr; frame_addr = frame_addr->previous) {
+    for (; frame_addr; frame_addr = frame_addr->previous)
+    {
+        auto resolved_addr = stack_chunk ? stack_chunk->resolve(frame_addr) : frame_addr;
+        // If the resolved address is different from the original, then we know
+        // that stack_chunk has the desired frame. We can safely copy, and use.
+        // Otherwise, either stack_chunk is nullptr or the frame is not in there
+        // and we need to copy from the original address.
+        if (resolved_addr != frame_addr)
+        {
+            iframe = *(_PyInterpreterFrame *)resolved_addr;
```

Comment:
Can we do this differently to avoid the copy?

Reply:
Probably, yeah. Thanks for pointing this out.

Reply:
hmm, I actually prefer internalizing call to `stack_chunk->resolve()` in `Frame::read()`